src/dist/dist.rs

```diff
@@ -159,6 +159,24 @@ static TRIPLE_MIPS64_UNKNOWN_LINUX_GNUABI64: &str = "mips64-unknown-linux-gnuabi
 #[cfg(all(not(windows), target_endian = "little"))]
 static TRIPLE_MIPS64_UNKNOWN_LINUX_GNUABI64: &str = "mips64el-unknown-linux-gnuabi64";
 
+/// Some old Rust versions don't have v2 manifests, but they don't have point releases either,
```

Comment:
I would suggest splitting this commit into smaller constituent parts that make exactly one conceptual change (like "extract `convert_old_two_part_version()` into a separate function"). It should then also be obvious from that commit why it is necessary (in this case, it's not clear why you extracted this).

(As part of this, your commits should have a more precise name, unlike "Refactor `dist`" which is very general.)

Reply:
> (in this case, it's not clear why you extracted this)

I used this function a second time in a later commit. I have been rewriting the history very early on in this PR, and yes this might look like I can perfectly predict things, but of course this is not the case :)

If you have any ideas about how to split the commits better given that context, please don't hesitate to tell me :D
(and commit renaming is always easier than a full rebase)

Reply:
> I used this function a second time in a later commit.

Yup, that's completely fine and a good reason for extracting the function.